File: 1448-maximum-69-number/1448-maximum-69-number.cpp
class Solution {
public:
    int maximum69Number (int num) {
        std::string numStr = std::to_string(num);

    // Find and replace the first occurrence of '6' with '9'
    for (char &c : numStr) {
        if (c == '6') {
            c = '9'; // Replace the first '6' with '9'
            break;   // Only change the first occurrence
        }
    }
    return std::stoi(numStr);
    }
};

File: 1448-maximum-69-number/1448-maximum-69-number.ts
function maximum69Number(num: number): number {
  const strNum = num.toString();
  const index = strNum.indexOf("6");

  if (index === -1) {
    return num; 
  }

  const updatedNum = strNum.substring(0, index) + "9" + strNum.substring(index + 1);
  return Number(updatedNum);
}
